core/toolbox.js

```diff
@@ -160,18 +160,6 @@ Blockly.Toolbox.prototype.init = function() {
   this.HtmlDiv.setAttribute('dir', workspace.RTL ? 'RTL' : 'LTR');
   svg.parentNode.insertBefore(this.HtmlDiv, svg);
 
-  // Clicking on toolbox closes popups.
-  Blockly.bindEventWithChecks_(this.HtmlDiv, 'mousedown', this,
-      function(e) {
-        if (Blockly.utils.isRightButton(e) || e.target == this.HtmlDiv) {
-          // Close flyout.
-          Blockly.hideChaff(false);
-        } else {
-          // Just close popups.
-          Blockly.hideChaff(true);
-        }
```

Comment:
This looks like a change in behavior. By removing the if statement, the flyout will always close, even where it didn't before.

Reply:
That's what I thought too, but I think the first condition:
`if (Blockly.utils.isRightButton(e) || e.target == this.HtmlDiv)`
actually always returns true because e.target will always be this.HtmlDiv.

I'm honestly not exactly sure what this function was doing before. I tried clicking all over the categories but I couldn't figure it out.

Reply:
I'd like to keep the old behaviour.  I think you can add your new listener and also keep this one without any issues.

Reply:
> Here are the MDN docs for [Event.target](https://developer.mozilla.org/en-US/docs/Web/API/Event/target) and [Event.currentTarget](https://developer.mozilla.org/en-US/docs/Web/API/Event/currentTarget). And if you're curious, here's some information on [event bubbling and capture](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#Event_bubbling_and_capture), which is part of why the distinction matters.

Thanks for the links! That was a really interesting read :D